library/alloc/src/collections/btree/navigate.rs

```diff
@@ -88,6 +87,7 @@ where
             (_, Unbounded) => max_node.last_edge(),
         };
 
+        // This can't be used for safety. The `PartialOrd` impl can change between calls.
```

Comment:
This is btree's own `PartialOrd,` (defined in node.rs, until #81094 reorganizes it away) and it is about the only real safety we have. If the condition fails, the iterator would never hit the other end and run off the parent of the root node (which is None) without a safe unwrap - that last safety has been "optimized" away).

Reply:
Good catch. This is a merge conflict waiting to happen.